workers/generic-worker/main.go

```diff
@@ -193,7 +194,15 @@ func main() {
 		case IDLE_TIMEOUT:
 			logEvent("instanceShutdown", nil, time.Now())
 			if config.ShutdownMachineOnIdle {
-				host.ImmediateShutdown("generic-worker idle timeout")
+				// If running with worker-runner, send shutdown message so it can
+				// unregister the worker before shutting down. Otherwise, shut down directly.
+				if WorkerRunnerProtocol.Capable("shutdown") {
```

Comment:
I think we need to `AddCapability("shutdown")` to this function https://github.com/taskcluster/taskcluster/blob/main/workers/generic-worker/workerrunner.go#L81

Nice find on this `shutdown` message capability too 👍🏻 Thanks